manager/workflow_manager.go

```diff
@@ -346,3 +346,28 @@ func (r *ResourceManager) ListWorkflowTemplates(namespace string) (workflowTempl
 
 	return
 }
+
+func (r *ResourceManager) ArchiveWorkflowTemplate(namespace, uid string) (archived bool, err error) {
+	allowed, err := r.kubeClient.IsAuthorized(namespace, "delete", "argoproj.io", "workflow", "")
+	if err != nil {
+		return false, util.NewUserError(codes.Unknown, "Unknown error.")
+	}
+	if !allowed {
+		return false, util.NewUserError(codes.PermissionDenied, "Permission denied.")
+	}
+
+	workflowTemplate, err := r.workflowRepository.GetWorkflowTemplate(namespace, uid, 0)
+	if err != nil {
+		return false, util.NewUserError(codes.Unknown, "Unknown error.")
+	}
+	if err == nil && workflowTemplate == nil {
```

Comment:
No need to check if err == nil - it was always be nil here because of the above check.

Reply:
Updated this and other instances of this check.